### PR TITLE
[all] Fix build & lint problems

### PR DESCRIPTION
**Pull request by mykola-kobets-epam**

### CODE REVIEW

File: iamclient/iamclient.go [hunk 1]
```diff
@@ -160,6 +160,11 @@ func (client *Client) GetSystemID() (systemID string) {
 	return client.systemID
 }
 
+// GetCurrentNodeInfo returns info for current node.
```
Comment (al1img) [inline iamclient/iamclient.go:163]: Adding new API should be done in separate commit

File: unitconfig/unitconfig.go [hunk 2]
```diff
@@ -69,12 +77,22 @@ var ErrAlreadyInstalled = errors.New("already installed")
  **********************************************************************************************************************/
 
 // New creates new unit config instance.
-func New(cfg *config.Config, client Client) (instance *Instance, err error) {
+func New(cfg *config.Config, nodeInfoProvider NodeInfoProvider, client Client) (instance *Instance, err error) {
 	instance = &Instance{
-		client:         client,
-		unitConfigFile: cfg.UnitConfigFile,
+		client:                   client,
```
Comment (al1img) [inline unitconfig/unitconfig.go:82]: These changes also should be in seprarate commit.

File: iamclient/iamclient.go (hunk 1 above)
Comment (mykola-kobets-epam) [reply iamclient/iamclient.go:163]: done

File: unitconfig/unitconfig.go [hunk 3]
```diff
@@ -39,10 +39,18 @@ import (
...
+	client                   Client
+	curNodeID                string
+	curNodeType              string
+	unitConfigFile           string
+	unitConfig               cloudprotocol.UnitConfig
+	currentNodeConfigChannel chan cloudprotocol.NodeConfig
+	unitConfigError          error
+}
+
+// Client client unit config interface.
```
Comment (al1img) [inline unitconfig/unitconfig.go:51]: Wrong comment

File: unitconfig/unitconfig.go [hunk 4]
```diff
@@ -261,6 +289,10 @@ func (instance *Instance) handleNodeConfigStatus() {
 		if err := instance.client.SetNodeConfig(instance.unitConfig.Version, *nodeConfig); err != nil {
 			log.WithField("NodeID", nodeConfigStatus.NodeID).Errorf("Can't update node config: %v", err)
 		}
+
+		if nodeConfig.NodeID != nil && *nodeConfig.NodeID == instance.curNodeID {
```
Comment (al1img) [inline unitconfig/unitconfig.go:293]: Not the right condition, let's discuss. Also should be called on update config.

File: iamclient/iamclient_test.go [hunk 5]
```diff
@@ -378,6 +378,38 @@ func TestGetNodeInfo(t *testing.T) {
...
+		IAMPublicServerURL:    publicServerURL,
+	}, &testSender{}, nil, true)
+	if err != nil {
+		t.Fatalf("Can't create IAM client: %s", err)
+	}
+	defer client.Close()
+
+	nodeInfo, err := client.GetCurrentNodeInfo()
+	if err != nil {
+		t.Errorf("Error is not expected: %s", err)
```
Comment (al1img) [inline iamclient/iamclient_test.go:401]: %v

File: iamclient/iamclient_test.go [hunk 6]
```diff
@@ -378,6 +378,38 @@ func TestGetNodeInfo(t *testing.T) {
...
+
+	defer publicServer.close()
+	defer protectedServer.close()
+
+	client, err := iamclient.New(&config.Config{
+		IAMProtectedServerURL: protectedServerURL,
+		IAMPublicServerURL:    publicServerURL,
+	}, &testSender{}, nil, true)
+	if err != nil {
+		t.Fatalf("Can't create IAM client: %s", err)
```
Comment (al1img) [inline iamclient/iamclient_test.go:395]: %v

File: iamclient/iamclient_test.go [hunk 7]
```diff
@@ -378,6 +378,38 @@ func TestGetNodeInfo(t *testing.T) {
 	}
 }
 
+func TestGetCurrentNodeInfo(t *testing.T) {
+	publicServer, protectedServer, err := newTestServer(publicServerURL, protectedServerURL)
+	if err != nil {
+		t.Fatalf("Can't create test server: %s", err)
```
Comment (al1img) [inline iamclient/iamclient_test.go:384]: %v

File: iamclient/iamclient_test.go (hunk 5 above)
Comment (mykola-kobets-epam) [reply iamclient/iamclient_test.go:401]: done

File: iamclient/iamclient_test.go (hunk 6 above)
Comment (mykola-kobets-epam) [reply iamclient/iamclient_test.go:395]: done

File: iamclient/iamclient_test.go (hunk 7 above)
Comment (mykola-kobets-epam) [reply iamclient/iamclient_test.go:384]: done

File: unitconfig/unitconfig.go (hunk 4 above)
Comment (mykola-kobets-epam) [reply unitconfig/unitconfig.go:293]: fixed

File: unitconfig/unitconfig.go (hunk 3 above)
Comment (mykola-kobets-epam) [reply unitconfig/unitconfig.go:51]: fixed

**Review by al1img — APPROVED** [review]

**Review by mlohvynenko — APPROVED** [review]
